core/templates/dev/head/forms/formBuilder.js

```diff
@@ -747,9 +728,313 @@ oppia.config(['$provide', function($provide) {
         });
 
         modalDialog.result.then(onSubmitCallback, onDismissCallback);
+        // The refocusFn arg is a function that restores focus to the text
+        // editor after exiting the modal, and moves the cursor back to
+        // where it was before the modal was opened.
         // 'finally' is a JS keyword. If it is just used in its ".finally" form,
         // the minification process throws an error.
         modalDialog.result['finally'](refocusFn);
+      }
+    };
+  }
+]);
+
+// Dynamically generate CKEditor widgets for the rich text components.
+oppia.run([
+  '$timeout', '$compile', '$rootScope', 'rteHelperService', 'oppiaHtmlEscaper',
+  function($timeout, $compile, $rootScope, rteHelperService, oppiaHtmlEscaper) {
+    var _RICH_TEXT_COMPONENTS = rteHelperService.getRichTextComponents();
+    _RICH_TEXT_COMPONENTS.forEach(function(componentDefn) {
+      // The name of the CKEditor widget corresponding to this component.
+      var ckName = 'oppia' + componentDefn.name;
```

Comment:
I think a similar construction is done elsewhere, in line 903. This sets up the potential for skew. May want to extract to a common method, or (perhaps better) make them a backend property.